core/planner.c

```diff
@@ -382,15 +382,31 @@ struct gaschanges {
 	int gasidx;
 };
 
-static struct gaschanges *analyze_gaslist(struct diveplan *diveplan, struct dive *dive, int *gaschangenr, int depth, int *asc_cylinder)
+// Return new setpoint if cylinderi is a setpoint change an 0 if not
+
+static int setpoint_change(struct dive *dive, int cylinderid)
+{
+	cylinder_t *cylinder = get_cylinder(dive, cylinderid);
+	if (cylinder->type.description[0] == 'S' &&
```

Comment:
Not a C-programmer, but I believe that `strncmp(cylinder->type.description, "SP ", 3) == 0` would be more idiomatic and also not invoke undefined behavior, should the length of `cylinder->type.description` be less than three.

Can `cylinder->type.description` ever be null?